src/storage/txn/store.rs

```diff
@@ -53,11 +53,18 @@ pub trait Scanner: Send {
     fn next(&mut self) -> Result<Option<(Key, Value)>>;
 
     /// Get the next [`KvPair`](KvPair)s up to `limit` if they exist.
-    fn scan(&mut self, limit: usize) -> Result<Vec<Result<KvPair>>> {
+    fn scan(&mut self, limit: usize, sample_step: usize) -> Result<Vec<Result<KvPair>>> {
+        let mut row_count = 0;
         let mut results = Vec::with_capacity(limit);
         while results.len() < limit {
             match self.next() {
                 Ok(Some((k, v))) => {
+                    if sample_step > 0 {
+                        if row_count % sample_step != 0 {
+                            continue;
+                        }
+                        row_count += 1;
```

Comment:
Should `row_count` be updated before the branch check?

Now, once `row_count % sample_step != 0` is true, `row_count` will never be changed and all other rows will be skipped.